clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusV2ClusterCachingAgent.java

```diff
@@ -0,0 +1,636 @@
+    imageCache.getRelationships().computeIfAbsent(SERVER_GROUPS.ns, key -> new HashSet<>()).add(data.serverGroupKey);
+    images.put(data.imageKey, imageCache);
+  }
+
+  @Override
+  public Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    Set<String> keys = providerCache.getIdentifiers("onDemand").stream()
+      .filter(it -> {
+        Map<String, String> key = Keys.parse(it);
+        return key != null &&
+          key.get("type").equals(SERVER_GROUPS.ns) &&
+          key.get("account").equals(account.getName()) &&
+          key.get("region").equals(region.getName());
+      })
+      .collect(Collectors.toSet());
+    return fetchPendingOnDemandRequests(providerCache, keys);
+  }
+
+  private Collection<Map> fetchPendingOnDemandRequests(ProviderCache providerCache, Collection<String> keys) {
+    return providerCache.getAll("onDemand", keys, RelationshipCacheFilter.none()).stream()
+      .map(it -> {
+        Map<String, Object> result = new HashMap<>();
+        result.put("id", it.getId());
+        result.put("details", Keys.parse(it.getId()));
+        result.put("cacheTime", it.getAttributes().get("cacheTime"));
+        result.put("proccessedCount", it.getAttributes().get("processedCount"));
+        result.put("processedTime", it.getAttributes().get("processedTime"));
+        return result;
+      })
+      .collect(Collectors.toList());
+  }
+
+
+  @Override
+  public Map pendingOnDemandRequest(ProviderCache providerCache, String id) {
+    Collection<Map> pendingOnDemandRequests = fetchPendingOnDemandRequests(providerCache, Collections.singletonList(id));
+    return pendingOnDemandRequests.isEmpty()
+      ? Collections.emptyMap()
+      : pendingOnDemandRequests.stream().findFirst().get();
+  }
+
+  private Map<String, CacheData> createCache() {
+    return new HashMap<>();
+  }
+
+  private String getAwsAccountId(String account, String region) {
+    return awsLookupUtil.get().awsAccountId(account, region);
+  }
+  private String getAwsVpcId(String account, String region) {
+    return awsLookupUtil.get().awsVpcId(account, region);
+  }
+
+  static class MutableCacheData implements CacheData {
+
+    final String id;
+
+    int ttlSeconds = -1;
+    final Map<String, Object> attributes = new HashMap<>();
+    final Map<String, Collection<String>> relationships = new HashMap<>(); //default value needs to be empty set!
+    public MutableCacheData(String id) {
+      this.id = id;
+    }
+    @Override
+    public String getId() {
+      return id;
+    }
+    @Override
+    public int getTtlSeconds() {
+      return ttlSeconds;
+    }
+
+    @Override
+    public Map<String, Object> getAttributes() {
+      return attributes;
+    }
+
+    @Override
+    public Map<String, Collection<String>> getRelationships() {
+      return relationships;
+    }
+
+    @JsonCreator
+    public MutableCacheData(@JsonProperty("id") String id,
```

Comment:
Let's generally keep class formatting uniform; constructors at the top after properties.